internal/metrics/recorder.go

```diff
@@ -270,7 +289,7 @@ func (r *Recorder) RecordAddonMetrics(addon *addonsv1alpha1.Addon) {
 	}
 }
 
-func (r *Recorder) recordAddonHealthInfo(addon *addonsv1alpha1.Addon) {
+func (r *Recorder) recordAddonHealthInfo(addon *addonsv1alpha1.Addon, unschedulableAddonPods *corev1.PodList) {
```

Comment:
Instead of resource types, we can enforce controllers to pass a custom reason which would make the recorder more generic and custom checks can take place out of the health info recorder as well.

```suggestion
func (r *Recorder) recordAddonHealthInfo(addon *addonsv1alpha1.Addon, reason string) {
```

Reply:
I have updated this by using a generic AddonHealth interface where details about the addon can be hidden and only provide "health information".